Revise OwnTracks region handling, support iBeacons.

pavoni opened the pull request:
This PR re-works OwnTracks entry / leave event handling.

The previous approach needed to be enabled with `use_events` which then disabled normal location tracking. 

This takes another approach allowing both`Location events` and `Region events` to be used together. It does this by storing a state to indicate a device is in a region - and using this to temporarily disable `Location` events until you exit the region.

There is also revised logic on exiting a region to attempt to update the location based on gps location, and only setting state to `Away` if this didn't already change the current location. This works well with iBeacons inside a larger Home area.

There is a small change to allow regions of size '0' not to be picked as priority by location code. This means you can have 'beacon only' zones, which still have a defined location.

~~The odd thing about the code is using a special state. Open to re-work this another way! I have in mind this could be extended to allow more stanes to support more iBeacon functions (eg are you close to your keys).~~ Revised to use a dictionary (thanks @balloob ).

There is a lot of change here - so would be grateful for some feedback - and happy to re-work.

Commit:
- add ds18S20 1-Wire sensor support
- gitignore emacs backup files

Commit:
little bug

deisi committed:
gitignore ready for emacs

deisi committed:
Update .gitignore

deisi committed:
Changed handling of config file 

I tried to implement your suggesteions for the default handlig of the device names. I think this way, everything you wanted is in.

deisi committed:
Improved onewire configuration

deisi committed:
merged with current dev

deisi committed:
updated with remote

deisi committed:
respect pylint suggestions

deisi committed:
respect flake8 errors

deisi committed:
new try to add everythin from the stylecheckers

auchter committed:
thermostat: move fan attribute up to thermostat

auchter committed:
thermostat: add service to control fan mode

auchter committed:
nest: implement fan control

balloob committed:
Merge pull request #883 from balloob/dev

0.11.0

balloob committed:
Enforce entity attribute types

balloob committed:
Fix MQTT reconnecting

Commit:
Add LIFX bulb support

Commit:
Remove use of warn()

Commit:
Attempt to fix ungrouped-imports pylint error

molobrakos committed:
fix issue where sensors and switches were duplicated because of component getting initialized twice. closes #913

balloob committed:
Allow forcing MQTT protocol v3.1

balloob committed:
Version bump to 0.11.1

MartinHjelmare committed:
Fix missing binary sensor types

* Add missing binary sensor types to sensor/mysensors.
* Remove unneeded pylint disable.

balloob committed:
Merge pull request #935 from balloob/0.11.1

0.11.1

balloob committed:
Allow passive zones

pavoni committed:
Revise region handling, support iBeacons.

Tidy.

Don't set location to NOT_HOME when exiting a beacon region.

Ignore zero size zones for GPS location priority.

Improve events lofic, handle iBeacons.

Add locks around critical sections.

Fix typo

Revise locking.

Reduce traces.

pavoni committed:
Revise region handling, support iBeacons.

Tidy.

Don't set location to NOT_HOME when exiting a beacon region.

Ignore zero size zones for GPS location priority.

Improve events lofic, handle iBeacons.

Add locks around critical sections.

Fix typo

Revise locking.

Reduce traces.

trollkarlen committed:
smtp: make smtp component thread safe

Also fix so it does not require ip, port, username and password when using local smtp server.
Add debug config.

Signed-off-by: Robert Marklund <robbelibobban@gmail.com>

turbokongen committed:
Added support for Dimming with lights in Rfxtrx module

turbokongen committed:
Small intendation fix

turbokongen committed:
Fixing of various test errors

turbokongen committed:
flake8 complaint fix

turbokongen committed:
Another flake8 fix too long line

molobrakos committed:
only query artwork by track_id if id is available (7.7 vs 7.9 version issue?)

molobrakos committed:
style

molobrakos committed:
style

molobrakos committed:
should be _id

pavoni committed:
Tidy.

pavoni committed:
Tidy.

sdague committed:
fix typo in log message

The plex component logs an htts url, which is confusing to people, as
they think something is broken, when it is not.

Closes #959

Commit:
Separate LIFX code and HA component

Commit:
Bump version liffylights

balloob committed:
Merge pull request #977 from sdague/dev

fix typo in log message

rmkraus committed:
Created automation decorator prototype

Created an initial iteration of an Automation decorator.

Commit:
Update liffylights version

Commit:
Fixes for lint

Commit:
Import 3rd party library inside method

Commit:
Update requirements

balloob committed:
Upgrade PyChromecast version

balloob committed:
Merge pull request #973 from turbokongen/dev

Added support for Dimming with lights in Rfxtrx module

balloob committed:
Merge pull request #971 from trollkarlen/smtp-fix

smtp: make smtp component thread safe

balloob committed:
Merge pull request #964 from balloob/passive-zones

Allow passive zones

balloob committed:
Merge pull request #980 from balloob/upgrade-cast

Upgrade PyChromecast version

Commit:
Add to .coveragerc

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -31,17 +38,25 @@ def owntracks_location_update(topic, payload, qos):
+        dev_id, host_name = _parse_topic(topic)
+
+        # Block updates if we're in a region
+        LOCK.acquire()
```
Please use `LOCK` as a context:

```python
with LOCK:
    …
```

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +76,111 @@ def owntracks_event_update(topic, payload, qos):
+
+    def _block_updates(dev_id, location):
+        """ Add owntracks region to block location updates """
+        hass.states.set(
```
This should not be set as an entity. Instead, define it as a variable scoped to the `setup` method and assign it here using `nonlocal` (see `gracefully_closed` in [this example](https://github.com/balloob/home-assistant/blob/dev/homeassistant/components/api.py#L110-L141)).

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +76,111 @@ def owntracks_event_update(topic, payload, qos):
+                     location, entry_location)
+        return False
+
+    def _block_updates(dev_id, location):
```
It seems that location data is not used at all. Why not keep track of all this data in a dict mapping `dev_id` => `bool`?

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +76,111 @@ def owntracks_event_update(topic, payload, qos):
+                   state.state != '')
+        return blocked
+
+    def _parse_topic(topic):
```
This is a pure function and can live outside of the `setup` scope.

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +76,111 @@ def owntracks_event_update(topic, payload, qos):
-            kwargs['gps_accuracy'] = data['acc']
+        dev_id = '{}_{}'.format(parts[1], parts[2])
+        host_name = parts[1]
+        return(dev_id, host_name)
```
```python
return dev_id, host_name
```

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +76,111 @@ def owntracks_event_update(topic, payload, qos):
+
+            see(**kwargs)
+
+            _block_updates(dev_id, '')
```
If you move to a dict approach, just popping the key-value pair out of the dict is neater.

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +76,111 @@ def owntracks_event_update(topic, payload, qos):
 
         elif data['event'] == 'leave':
-            location = STATE_NOT_HOME
+            if not _valid_leave(dev_id, location):
```
Can you inline this method as it is only called once? That way it is more clear to the reader of this code what is going on.

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +76,111 @@ def owntracks_event_update(topic, payload, qos):
+        else:
+            location = data['desc']
+
+        kwargs = {
```
Let's extract this into a method so that both times the kwargs are rendered the same. For example, you forgot the battery attribute here.

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -31,17 +38,25 @@ def owntracks_location_update(topic, payload, qos):
+        # Block updates if we're in a region
+        LOCK.acquire()
+        if _is_blocked(dev_id):
+            _LOGGER.info("Owntracks update rejected - inside region")
```
I would consider this message a log level `debug`

balloob committed:
Load YAML config into an ordered dict

balloob committed:
Move split_entity_id to helpers

balloob committed:
Move extract_entity_id to service helpers

balloob committed:
Move generate_entity_id to entity helpers

balloob committed:
Merge pull request #981 from balloob/ordered-yaml

Load YAML config into an ordered dict

Commit:
Updated the entity names for all z-wave devices.
Z-Wave have a unique node id, this is now added to the entity name.

This is a change will break old configs.  The Z-Wave entity_id will have to be updated in configurations.

The reason for this change is that the old behavior created duplicate entity_ids if you have several similar devices. The order devices is included is not consistent so when the service was restarted the entity ids could change and the configuration for entities will be invalid.

There was also inconsistencies between the naming for sensors and switches.
(cherry picked from commit 33a78df)

turbokongen committed:
Merge pull request #1 from balloob/dev

Updating base

pavoni committed:
Tidy after feedback.

pavoni committed:
Tidy after feedback.

Commit:
Fix wrongly generated requirements

Commit:
Fix wrongly generated requirements

pavoni committed:
Use dictionary for regions instead of state machine, handle nested regions.

pavoni committed:
Use dictionary for regions instead of state machine, handle nested regions.

pavoni commented:
@balloob Good call on the local dictionary. I've also improved the code so that it can handle nested regions (ie if you enter region `big` and then region `small` when you exit `small` it puts you back in `big`. Of course it's handles it OK if you exit `big` first!.

turbokongen committed:
Send correct command to pyrfxtrx

Although it seems to work with send_on, it throws an logged error. Using correct command against pyrfxtrx removes this error.

turbokongen committed:
flake8 complaint fix

balloob committed:
Merge pull request #986 from turbokongen/dev

Small errorlog fix.

balloob committed:
Update version pynetgear

balloob committed:
Merge pull request #987 from balloob/update-pynetgear

Update version pynetgear

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -8,16 +8,26 @@
-CONF_TRANSITION_EVENTS = 'use_events'
+REGIONS_ENTERED = defaultdict(list)
+
+STATE_OWNTRACKS_LAST_LOCATION = 'owntracks.location_{}'
```
These are no longer used now.

balloob commented:
This component is getting quite complex now, I would like to see some tests to verify correct behavior.

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -31,25 +41,24 @@ def owntracks_location_update(topic, payload, qos):
-        see(**kwargs)
+        # Block updates if we're in a region
+        with LOCK:
+            region = _in_region(dev_id)
```
This method is only used once and basically boils down to this:

```python
if REGIONS_ENTERED[dev_id]:
```

I think we can remove that method.

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -31,25 +41,24 @@ def owntracks_location_update(topic, payload, qos):
+        with LOCK:
+            region = _in_region(dev_id)
+            if region:
+                _LOGGER.info(
```
This should be a debug.

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +69,119 @@ def owntracks_event_update(topic, payload, qos):
+            with LOCK:
+                if zone is not None:
+                    kwargs['location_name'] = location
+                    _enter_region(dev_id, location)
```
Please inline this function as it is only used once.

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +69,119 @@ def owntracks_event_update(topic, payload, qos):
-            location = STATE_NOT_HOME
+            current_location = hass.states.get(
+                "device_tracker.{}".format(dev_id)).state
+            new_region = _exit_region(dev_id, location)
```
Let's inline this function too.

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +69,119 @@ def owntracks_event_update(topic, payload, qos):
+            if not new_region:
+                _LOGGER.info("Exit from %s to GPS", location)
+                see(**kwargs)
+                _force_state_change_if_needed(dev_id, location, kwargs)
```
Inline too.

balloob commented on homeassistant/components/zone.py:
```diff
@@ -44,6 +44,7 @@ def active_zone(hass, latitude, longitude, radius=0):
         within_zone = zone_dist - radius < zone.attributes[ATTR_RADIUS]
         closer_zone = closest is None or zone_dist < min_dist
         smaller_zone = (zone_dist == min_dist and
+                        zone.attributes[ATTR_RADIUS] != 0 and
```
Why is this check needed? A person can never be in this zone because `within_zone` will always be False.

pavoni replied:
It seemed to be picking the smallest region associated with the GPS position .

balloob replied:
Yep, that's by design.

balloob replied:
Why would anyone even create zones with radius 0 ?

pavoni replied:
GPS isn't precise for small regions - so you can't distinguish between (say) rooms.

But with ibeacons you can be precise about where you are. However you then you don't want gps picking the region (because it'll get it wrong).

So radius 0 is an attempt to say 'don't pick this region via gps'. But it didn't work without the check I added. There may be better fixes.

It would also be nice to be able to have room zones marked as 'consider home', so you can nest iBeacon zones for rooms inside a larger home zone.

balloob replied:
You mean something like a passive zone that is not to be picked up by GPS #964 :wink:

pavoni replied:
Perfect I'll delete this hack!

Still good to have 'consider home' for beacon zones.

auchter committed:
thermostat: split up services

pavoni committed:
Inline region membership functions.

pavoni committed:
Inline region membership functions.

pavoni commented:
Agreee on the complexity and some tests. Will add in the next few days.

pavoni committed:
Inline function.

pavoni committed:
Inline function.

rmkraus committed:
Reverting Automation decorator in favor of a new approach.

rmkraus committed:
Moved sunrise/sunset tracking to helpers

The automation component contained some very handy and generic
functions for tracking sunset and sunrise. This was moved to
helpers/event.py.

rmkraus committed:
Initial pass at event decorators

Created event decorators for custom components. Decorators were created
for the events: track_state_change, track_sunrise, track_sunset, and
track_time_change.

rmkraus committed:
Fixed bugs to allow HA to boot again

1) helpers/event should not import the sun component unless it is
requested. This prevents circular import.
2) fixed import typo in bootstrap
2) bootstrap cannot import event_decorators until it is needed because
this leads to a circular import.

molobrakos committed:
Merge pull request #990 from molobrakos/squeezebox-fix

bugfix for squeezebox cover art regression

rmkraus committed:
A few lint fixes to event decorators.

balloob committed:
Allow groups to be used as views

rmkraus committed:
Another revision on event decorators

This revision of event decorators removes much of the complexity. The
decorated functions are no longer wrapped with a class that tracks
last_run, etc. Bootstrap now gives hass to the event_decorators module
before initializing components so the decorators no longer require
activation.

rmkraus committed:
Many updates regarding event decorators

1. Added HASS to the arguments for callbacks that are created with
event decorators.
2. Added a service decorator.
3. Updated example.py in the example config to use the event decorators.

rmkraus committed:
Updated example.py component

Cleaned up example.py to better handle failed loads.

rmkraus committed:
Lint fixes and additions to event decorators

1. service decorator was overwriting the function name with one of its
arguments.
2. Accidentally left an extra argument in track_sunrise.
3. Added track_utc_time_change decorator.

rmkraus committed:
Added event helper tests

1. Added tests for all event decorators
2. Added tests for sunrise and sunset event helpers

Commit:
Add preliminary support for transition time

Commit:
Bump version of liffylights

Commit:
Fix long line

rmkraus committed:
Moved service decorator to service helpers

Moved the service decorator to the service helpers module and moved the
associated tests.

rmkraus committed:
Merge remote-tracking branch 'balloob/dev' into automation-decorator

# Conflicts:
#	homeassistant/helpers/service.py
#	tests/helpers/test_service.py

rmkraus committed:
Fixed lint issue from merge

extract_entity_ids from the service helpers was overwriting the service
decorator with one of its attributes. This was fixed.

rmkraus committed:
Removed service decorator from event decorators

rmkraus committed:
Removed decorator callback

The decorator callback was not actually necessary so it was removed and
replaced with a partial function instead.

balloob committed:
Merge pull request #979 from rmkraus/automation-decorator

Automation Decorator for custom components

balloob committed:
Merge pull request #992 from balloob/sections

Add support for sections in the frontend [Fixes #100705168]

balloob committed:
Merge pull request #988 from auchter/thermostat-fan

Add service to control a thermostat's fan

Commit:
Update requirements_all

balloob committed:
Hide groups when they are views or auto defined

Commit:
Update to v0.6 of liffylights

Commit:
Update requirements_all

Commit:
Update to v0.8.3 of liffylights

Commit:
Update requirements_all

Commit:
Add extra debugging
Add duration to power on

pavoni committed:
Disable lint (after inlining).

pavoni committed:
Disable lint (after inlining).

Commit:
Fix liffylights import

Commit:
Update requirements_all

pavoni committed:
Add tests for owntracks

pavoni committed:
Add tests for owntracks

pavoni committed:
Changed not needed.

pavoni committed:
Changed not needed.

Commit:
Update for new liffylights version

persandstrom committed:
update vsure version

Commit:
Update liffylights version

Commit:
Fix logging typo

persandstrom committed:
Merge pull request #995 from persandstrom/vsure0.4.8

update python-verisure to version 0.4.8

Commit:
Brightness from HA overrides brightness from HSV conversion

Commit:
bump orvibo version

MartinHjelmare committed:
Fix unit of measurement

* Add version check in unit_of_measurement, to avoid error.
* Remove not needed metric check in unit_of_measurement.
* Add more value types to unit_of_measurement.
* Fix unit prefix
* Remove unused SCAN_INTERVAL variable.

happyleavesaoc committed:
Merge pull request #994 from happyleavesaoc/orvibo_bump

Orvibo version bump

balloob committed:
Merge pull request #930 from TangoAlpha/dev

Add LIFX bulb support

balloob committed:
Update frontend

MartinHjelmare committed:
Fix adding devices on the fly

Devices were not added without persistence enabled and restart of HA.
Node id was added to defaultdict(list) by mistake when checking if list
of defaultdict was empty.

* Fix adding devices in mysensors_callback.
* Change devices to regular dict.

balloob committed:
Add views to demo

balloob committed:
Fix lint issue

balloob committed:
Update frontend with view url sync

balloob committed:
Merge pull request #1000 from balloob/view-url-sync

Update frontend with view url sync

balloob committed:
Merge branch 'master' into dev

Conflicts:
	homeassistant/const.py

balloob committed:
Fix Chrome url sync

fabaff committed:
Update docstring

Commit:
Updated teh entity_id generation

Entity ids will now look like this:
sensor.philio_technology_corporation_psm021_slim_multisensor_doorwindow_sensor_7
sensor.philio_technology_corporation_psm021_slim_multisensor_luminance_7
sensor.philio_technology_corporation_psm021_slim_multisensor_motion_sensor_7
sensor.philio_technology_corporation_psm021_slim_multisensor_tamper_sensor_7
sensor.philio_technology_corporation_psm021_slim_multisensor_temperature_7

sensor.philio_technology_corporation_psm021_slim_multisensor_doorwindow_sensor_8
sensor.philio_technology_corporation_psm021_slim_multisensor_luminance_8
sensor.philio_technology_corporation_psm021_slim_multisensor_motion_sensor_8
sensor.philio_technology_corporation_psm021_slim_multisensor_tamper_sensor_8
sensor.philio_technology_corporation_psm021_slim_multisensor_temperature_8

And if there is several values of the same type in a node the instance id will be appended like this (three switches in the same device):
switch.fibaro_system_unknown_type0202_id1002_switch_9
switch.fibaro_system_unknown_type0202_id1002_switch_9_2
switch.fibaro_system_unknown_type0202_id1002_switch_9_3

MagnusKnutas committed:
Running periodic tasks

MagnusKnutas committed:
hourly periodic task

MagnusKnutas committed:
test past midnight support for hours

MagnusKnutas committed:
Warnings for not divisible settings

MagnusKnutas committed:
Some warnings if input is wrong for periodic tasks

MagnusKnutas committed:
New and improved handling of the matching! Kudos to @balloob

MagnusKnutas committed:
Year periodic test (just to trigger travis but should maybe be here anyway)

MagnusKnutas committed:
Unused import was the problem

MagnusKnutas committed:
Remove warnings no longer needed (I think :D)

MartinHjelmare committed:
Merge pull request #999 from MartinHjelmare/fix-not-adding-devices

Fix adding devices on the fly

MagnusKnutas committed:
Sorry, I was wrong they are needed

MagnusKnutas committed:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	homeassistant/components/light/zwave.py
#	homeassistant/components/switch/zwave.py

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +67,101 @@ def owntracks_event_update(topic, payload, qos):
+                             location, current_location)
+                return
+
+            zone = hass.states.get("zone.{}".format(new_region))
```
Should you test if zone exists?

pavoni replied:
The zone was pulled from the list of zones you entered. So it wouldn't have been added if it didn't exist. Happy to check if you want to be defensive for the future.

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +67,101 @@ def owntracks_event_update(topic, payload, qos):
+    return dev_id, kwargs
+
+
+def _get_gps_from_zone(kwargs, zone):
```
The name of this method should probably be `_set_gps_from_zone`

balloob replied:
You are missing a doc string. I wonder why the linter didn't catch this

pavoni replied:
My guess is functions with underscores are assumed to be internal. But happy to add one.

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +67,101 @@ def owntracks_event_update(topic, payload, qos):
-        mqtt.subscribe(hass, LOCATION_TOPIC, owntracks_location_update, 1)
 
-    return True
+def _parse_see_args(topic, data):
```
Missing doc string

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +67,101 @@ def owntracks_event_update(topic, payload, qos):
+        kwargs['gps'] = (
+            zone.attributes['latitude'],
+            zone.attributes['longitude'])
+        kwargs['gps_accuracy'] = 1
```
This should be the accuracy of the zone, because we know the person is somewhere in that region.

pavoni replied:
Do you mean the radius of the zone? This makes sense but just want to confirm.

balloob replied:
Yes

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +67,101 @@ def owntracks_event_update(topic, payload, qos):
+                see(**kwargs)
+
+                # if gps location didn't set new state, force to away
+                current_location = hass.states.get(
```
When would it happen that it didn't set a new state and how would we be able to detect this without having to call `see` twice, because that seems like a hack.

pavoni replied:
Less of a hack than what was there before! (Which was to always set STATE_NOT_HOME whenever you exited a region! )

What I think the previous author was trying to do was to cope with the situation where you get a zone exit on the boundary of a zone. You should respect that OwnTracks has said the zone was exited.  However if the GPS location is on the boundary HA may stick you back in the zone.

So what I've tried to do is allow HA to respond sensibly - and fallback to what the old code did, if HA didn't exit the zone.

My experiments indicate this is questionable using beacons - because they can randomly disconnect -  if that happens a forced exit is probably wrong. So could just do this for gps exits.

If you want to keep comparability with the old code we need some sort of forced exit.

Purity is perhaps a suggestion to make your HA zone smaller than your OwnTracks one, and remove the forced exit.

I don't think there is a perfect solution here -we're trying to reconcile two incompatible data sources. So happy to go with whatever makes most sense.


As you probably know the issue is that the OwnTracks location tracking can have long gaps between updates. So if you incorrectly interpret an exit - the state could be wrong for a long time.

pavoni replied:
@balloob I had an idea. What about adding a config to each own tracks zone saying what to do when you exit. Default should just be to exit using GPS location. But you could specify 'not-home' to get the old behavior (or another zone if you are using nested zones).

balloob replied:
I would go for purity here. If OwnTracks tells that we are leaving a region, and we have no region to fall back to, we should just depend on the GPS. I don't think it is up to us to try to handle edge cases but have the user take care of that when he configures his zones.

MagnusKnutas committed:
Fixes bug in time trigger and adds test for ex. /two

balloob commented on homeassistant/components/device_tracker/owntracks.py:
```diff
@@ -60,47 +67,101 @@ def owntracks_event_update(topic, payload, qos):
 
         elif data['event'] == 'leave':
-            location = STATE_NOT_HOME
+            current_location = hass.states.get(
```
This also seems like a hack, OwnTracks should be able to keep track of it's own state of a device. Isn't that what regions is for ?

pavoni replied:
I didn't think is was safe to assume that no other code would set the state between an enter and leave event. Didn't you intend there to be multiple device trackers?  In any case couldn'tuser code or an automation something else have set the state after we did when we entered a zone?

MagnusKnutas committed:
pushed test code :) fixed now

MagnusKnutas committed:
Tests for time and a fix for wrong access of _LOGGER

MagnusKnutas committed:
And a test just to see if nothing works if not providing any vars

MagnusKnutas committed:
Remove not needed log warning

MagnusKnutas committed:
Dont wrap '/int' parameters with tuple

MagnusKnutas committed:
Remove unused import

Commit:
Moved common z-wave node functionality to a common z-wave class

fabaff committed:
Updgrade to astral 0.9

fabaff committed:
Upgrade influxdb to 2.11.0

fabaff committed:
Upgrade python-telegram-bot to 3.2.0

fabaff committed:
Upgrade blockchain to 1.2.1

fabaff committed:
Update docstrings

MartinHjelmare committed:
Fix device_state_attributes

* Return a dict with keys as the names of the enum members instead of
	the member values.
* This fixes a TypeError: unorderable types: int() < str()

MartinHjelmare committed:
Merge pull request #1010 from MartinHjelmare/fix-device-attr

Fix device_state_attributes

balloob committed:
Merge pull request #1009 from fabaff/docstrings

Update docstrings

balloob committed:
Merge pull request #1008 from fabaff/blockchain-upgrade

Upgrade blockchain to 1.2.1

balloob committed:
Merge pull request #1007 from fabaff/telegram-upgrade

Upgrade python-telegram-bot to 3.2.0

balloob committed:
Merge pull request #1006 from fabaff/influxdb-upgrade

Upgrade influxdb to 2.11.0

balloob committed:
Merge pull request #1005 from fabaff/astral-upgrade

Updgrade to astral 0.9

MartinHjelmare committed:
Merge pull request #1011 from MartinHjelmare/fix-unit-of-measurement

Fix unit of measurement

balloob committed:
Merge branch 'pr/78' into dev

Conflicts:
	homeassistant/const.py

balloob committed:
Merge pull request #1002 from kennedyshead/dev

Running periodic tasks

fabaff committed:
Add link to docs and update docstrings

Commit:
Updated Imports.
ZWave modules now import the domain from the base domain.

balloob committed:
Merge pull request #982 from stefan-jonasson/dev

Updated the entity names for all z-wave devices.

fabaff committed:
Add twitter docs

balloob committed:
Add onewire to coverage

fabaff committed:
Add link to docs and remove configuration details

fabaff committed:
Add link to docs

fabaff committed:
Update docstring

fabaff committed:
Add link to docs

fabaff committed:
Update docstring and fix typos

fabaff committed:
Move configuration details to docs

fabaff committed:
Add link to docs

fabaff committed:
Update docstings

fabaff committed:
Upgrade py-cpuinfo to 0.1.8

fabaff committed:
Upgrade psutil to 3.4.2

balloob committed:
Merge pull request #1016 from fabaff/cpuinfo-upgrade

Upgrade py-cpuinfo to 0.1.8

balloob committed:
Merge pull request #1017 from fabaff/psutil-upgrade

Upgrade psutil to 3.4.2

Commit:
Move Insteon API KEY into configuration.yaml

Commit:
Use validate_config, watch for import exception

pavoni committed:
Remove forces exit test. Add accuracy tests.

pavoni committed:
Remove forces exit test. Add accuracy tests.

pavoni committed:
Rename function, remove forces zone exit, revise accuracy.

pavoni committed:
Rename function, remove forces zone exit, revise accuracy.

molobrakos committed:
allow monitoring weather at other location than home (such as summer house)

balloob committed:
Merge pull request #1025 from molobrakos/yr-position

allow monitoring weather at other location than home

Commit:
Remove unnecessary try around import

Commit:
Add ability to control dimmable sources

balloob committed:
Merge pull request #1022 from haraldnagel/dev

Move Insteon API KEY into configuration.yaml

balloob committed:
Fix imports

balloob committed:
Merge pull request #1035 from balloob/fix-imports

Fix imports and rename insteon to insteon_hub

balloob commented:
It looks good. You can merge it when you're happy with it :dolphin:

pavoni committed:
Handle leading dash in owntracks region.

pavoni committed:
Handle leading dash in owntracks region.

pavoni committed:
Merge branch 'owntracks_beacon' of https://github.com/balloob/home-assistant into owntracks_beacon

pavoni commented:
Mess up this branch - will re-open a new PR.

balloob commented:
You can do things like revert and interactive rebase to fix messed up
branches. After that you force push to this branch and it would have been
all good

On Fri, Jan 29, 2016, 01:35 Greg Dowling <notifications@github.com> wrote:

> Closed #976 <https://github.com/balloob/home-assistant/pull/976>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/976#event-531042340>.
>
